File: 백준 문제 풀이 모음/class 3/11047.py
import sys

N, K = map(int, sys.stdin.readline().split())

coin = [0]
for i in range(N) :
    
    coin.append(int(sys.stdin.readline()))

count = 0
for i in range(len(coin)-1, 0, -1) :
    
    if K // coin[i] == 0 :
        continue
    
    if K == 0 :
        break
    count += (K // coin[i])
    K = K % coin[i]

print(count)

File: 백준 문제 풀이 모음/kotlin/class 3/11047.kt

fun main() = with(System.`in`.bufferedReader()){
    val str = readLine().split(' ').toList()
    val n = str[0].toInt()
    val k = str[1].toInt()

    val list = mutableListOf<Int>()
    repeat(n) {
        val input = readLine().toInt()
        list.add(input)
    }

    var count = 0
    var value = k
    for (i in n-1 downTo 0) {
        if (value / list[i] == 0) continue

        count += value / list[i]
        value = value % list[i]
    }

    println(count)
}
